File: 프로그래머스/lv2/42747. H－Index/H－Index.js
function solution(citations) {
    let i = 0; // i 이상인 논문의 개수
    
    for(i = citations.length; i >= 0; i--) {
        const overI = citations.filter(c => c >= i).length;
        if(overI >= i) break;
    }
    
    return i;
}

/*
i이상인 논문의 개수를 센다. 이 개수는 overI라고 한다. 
이 overI가 i 보다 크면 멈춘다. 그게 조건이니까

*/

File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    if len(citations) == 1:
        ans = 0 if citations[0] == 0 else 1
        return ans

    ans = 0
    sorted_c = sorted(citations, reverse=True)
    
    for i in range(len(sorted_c)):            
        if(sorted_c[i] < i + 1):
            return i

    return len(sorted_c)  